library/std/src/io/mod.rs

```diff
@@ -1385,17 +1385,18 @@ pub trait Write {
     ///
     /// This function will attempt to write the entire contents of `buf`, but
     /// the entire write might not succeed, or the write may also generate an
-    /// error. A call to `write` represents *at most one* attempt to write to
+    /// error. Typically, a call to `write` represents one attempt to write to
     /// any wrapped object.
```

Comment:
```suggestion
    /// any wrapped object. At most one underlying write operation
    /// (whether successful or failed) can occur involving the new
    /// bytes supplied by the caller, since once that operation
    /// occurs, the return value would need to reflect that success
    /// or error.
```

Reply:
I dunno if using strong language here is needed.  Current wording around Ok(n) and Ok(0) should steer all sane implementations towards doing at most one write for new bytes already.  Meanwhile, what if I implement a `struct MultiplexWrite<U: Write, V: Write>(U, V);` which sends single write to two underlying writers? Or an `AutoReconnectWriter(TCPStream)` which automatically re-establishes connection on connection closed errors?